docs/architecture/deep-dives/autofill/autofill-menu.md

```diff
@@ -0,0 +1,299 @@
+---
+sidebar_position: 5
+---
+
+# Inline Auto-fill Menu
+
+The Inline Auto-fill Menu allows users to access and autofill credentials from their Bitwarden vault
+directly within webpage forms. Enabled by default with the installation of the Bitwarden extension,
+this feature can be managed through the autofill settings within the Bitwarden extension. When
+active, the extension injects the Auto-fill Menu into the webpage's DOM via the content script
+responsible for enabling the autofill functionality.
+
+## Project Structure
+
+The core scripts for the Auto-fill Menu feature are located within the `/apps/browser/autofill`
+directory of the [`clients` repository](https://github.com/bitwarden/clients). These files form the
+backbone of the feature's functionality.
+
+| File                                  | Responsibility                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |
+| ------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `overlay.background.ts`               | Orchestrates communication among three key components: the content script containing the`AutofillOverlayContentService`, the extension pages displaying the Auto-fill Menu UI, and the extension background script. It facilitates the activation of various logic sequences by transmitting messages between these components.                                                                                                                                                                                          |
```

Comment:
💭 In general is there value to linking more of these code components? The service, class, etc. There are some links like this further down and it would be good to be consistent.

Reply:
Good call, I've updated this section to link to the repo source.